core/src/main/scala/org/apache/spark/deploy/worker/DriverRunner.scala

```diff
@@ -48,13 +48,16 @@ private[deploy] class DriverRunner(
     val securityManager: SecurityManager)
   extends Logging {
 
-  @volatile private var process: Option[Process] = None
-  @volatile private var killed = false
+  private var workerThread: Thread = null
+  private var process: Process = null
```

Comment:
I opted not to use an AtomicReference on the `process` variable, although it is shared with the ShutdownHookManager, I think if the hook is invoked then it's not an issue at that point because it means the JVM is exiting right?